src/ProjectTemplates/test/MvcTemplateTest.cs

```diff
@@ -247,7 +247,7 @@ private async Task MvcTemplateBuildsAndPublishes(string auth, string[] args)
             Assert.True(0 == createResult.ExitCode, ErrorMessages.GetFailedProcessMessage("create/restore", Project, createResult));
 
             // Verify building in debug works
-            var buildResult = await Project.RunDotNetBuildAsync();
+            var buildResult = await Project.RunDotNetBuildAsync(additionalArgs: "/p:NoWarn=NU1605");
```

Comment:
Strange this still occurs. What was the exact warning❔

Reply:
I'm trying this out as a workaround.

The exact error is:

```
version 16.7.0-preview-20330-02+1eab2845b for .NET\r\nCopyright (C) Microsoft Corporation. All rights reserved.\r\n\r\nF:\workspace\_work\1\s\.dotnet\sdk\5.0.100-preview.7.20330.3\MSBuild.dll -maxcpucount -property:Configuration=Release -target:Publish -verbosity:m /bl .\AspNet.pm24ssyvh1s.sln\r\n  You are using a preview version of .NET. See: https://aka.ms/dotnet-core-
preview\r\nF:\workspace\_work\1\s\src\ProjectTemplates\BlazorTemplates.Tests\bin\Release\net5.0\TestTemplates\AspNet.pm24ssyvh1s\Server\AspNet.pm24ssyvh1s.Server.csproj : error NU1605: Detected package downgrade: Microsoft.AspNetCore.Authentication.JwtBearer from 5.0.0-preview.6.20312.15 to 5.0.0-ci. Reference the package directly from the project to select a different version. 
\r\nF:\workspace\_work\1\s\src\ProjectTemplates\BlazorTemplates.Tests\bin\Release\net5.0\TestTemplates\AspNet.pm24ssyvh1s\Server\AspNet.pm24ssyvh1s.Server.csproj : error NU1605:  AspNet.pm24ssyvh1s.Server -> Microsoft.Identity.Web 0.2.1-preview -> Microsoft.AspNetCore.Authentication.JwtBearer (>= 5.0.0-preview.6.20312.15) 
\r\nF:\workspace\_work\1\s\src\ProjectTemplates\BlazorTemplates.Tests\bin\Release\net5.0\TestTemplates\AspNet.pm24ssyvh1s\Server\AspNet.pm24ssyvh1s.Server.csproj : error NU1605:  AspNet.pm24ssyvh1s.Server -> Microsoft.AspNetCore.Authentication.JwtBearer (>= 5.0.0-ci)\r\nF:\workspace\_work\1\s\src\ProjectTemplates\BlazorTemplates.Tests\bin\Release\net5.0\TestTemplates\AspNet.pm24ssyvh1s\Server\AspNet.pm24ssyvh1s.Server.csproj : error NU1605: Detected package downgrade: Microsoft.AspNetCore.Authentication.OpenIdConnect from 5.0.0-preview.6.20312.15 to 5.0.0-ci. Reference the package directly from the project to select a different version. 
\r\nF:\workspace\_work\1\s\src\ProjectTemplates\BlazorTemplates.Tests\bin\Release\net5.0\TestTemplates\AspNet.pm24ssyvh1s\Server\AspNet.pm24ssyvh1s.Server.csproj : error NU1605:  AspNet.pm24ssyvh1s.Server -> Microsoft.Identity.Web 0.2.1-preview -> Microsoft.AspNetCore.Authentication.OpenIdConnect (>= 5.0.0-preview.6.20312.15) 
\r\nF:\workspace\_work\1\s\src\ProjectTemplates\BlazorTemplates.Tests\bin\Release\net5.0\TestTemplates\AspNet.pm24ssyvh1s\Server\AspNet.pm24ssyvh1s.Server.csproj : error NU1605:  AspNet.pm24ssyvh1s.Server -> 
Microsoft.AspNetCore.Authentication.OpenIdConnect (>= 5.0.0-ci)\r\n  You are using a preview version of .NET. See: https://aka.ms/dotnet-core-preview\r\n  You are using a preview version of .NET. See: https://aka.ms/dotnet-core-preview\r\n  AspNet.pm24ssyvh1s.Shared -> 
```

I spoke with @pranavkm and he mentioned that the build might be failing in our public builds might be failing because they don't have access to the preview8 packages in the internal feed. When this happens, they fallback to the local verifications even though we have the `VersionPrefix` check in our build.

I tried this out by submitting an internal PR but it failed with the same reason above. I'm not sure what the issue is here.

I disabled the warning as a last ditch effort since this change is a workaround that we will remove after the preview8 release and I didn't want to overinvest in it.

If you have ideas of cheap fixes for this, let me know.

Reply:
OIC the problem is the `Condition`s in https://github.com/dotnet/aspnetcore/pull/24600/files#diff-a02e9b40ed5c9908719840e31ded1de3R45-R46 See suggestion there

Reply:
I'm pretty sure I had tried a similar condition at some point last night but I'll give this a go.